cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -412,6 +419,81 @@ func (p *BicepProvider) destroyResourceGroups(
 	return nil
 }
 
+func (p *BicepProvider) purgeItems(
+	ctx context.Context,
+	asyncContext *async.InteractiveTaskContextWithProgress[*DestroyResult, *DestroyProgress],
+	keyVaults []*azcli.AzCliKeyVault,
+	appConfigs []*azcli.AzCliAppConfig,
```

Comment:
I think this is fine, but I wonder if we could create an interface for all `Purgeable` resources and have a single list of resources that we operate on.